### PR TITLE
Base find

### LINKED ISSUE #413: Implement basefind feature
Basefind aims at find the base address of a raw binary by using ASCII strings and their intersections.

There are various implementations of this:
- https://github.com/sgayou/rbasefind (in Rust)
- https://github.com/mncoppola/ws30/blob/master/basefind.py (Python)
- https://github.com/mncoppola/ws30/blob/master/basefind.cpp (C++)
- https://gist.github.com/rpw/2c4064712638bce602755a938991e5e9 (Python)

Since Rizin is often used for reversing raw binaries (e.g. firmware) it's commonly used feature in this scenario. It could be a part of the Rizin itself, moreover, reusing the strings detection logic, see https://github.com/rizinorg/rizin/issues/38

### PR DESCRIPTION
**Pull request by gnole**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Implement basefind. 
command: `rz-bin -Y file`

**Test plan**

CI is green

**Closing issues**

Closes https://github.com/rizinorg/rizin/issues/413

### CODE REVIEW

**Comment by gnole** [comment]
RizinBasefind

**Comment by gnole** [comment]
TODO Coding style/Rewrite code from C++ to C.
TODO Implement multithreading

File: librz/core/rz_basefind.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
+//MIT Copyright (c) 2013 Michael Coppola
```
Comment (XVilka) [inline librz/core/rz_basefind.cpp:1]: Use SPDX

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/rz_basefind.cpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+#include <map>
+#include <unordered_set>
+#include <vector>
+#include <iostream>
+static const size_t minLength = 10;
+static const size_t WORD_LEN = 4; //32-bit assumed in a few places below
+typedef uint32_t offset_t;
+typedef std::vector<std::pair<offset_t, unsigned>> ptable; //pointer table
+typedef std::unordered_set<offset_t> stable; //string table
+typedef std::vector<std::pair<offset_t, unsigned>> scorecard;
```
Comment (wargio) [inline librz/core/rz_basefind.cpp:34]: for this you can use
```c
typedef rz_basefinds_t {
     offset_t offset;
     ut32 score;
} RzBasefinds;
```
The all `std::vector` and `std::unordered_set` you can use `rz_pvector` and use the `rz_pvector_find` to check if the set has the address.

**Review by wargio — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
Please reopen if you plan to continue the development. For now, I am closing as this is far from complete and stalled.

**Pull request closed without merging**